frontend/pages/admin/IntegrationsPage/cards/MdmSettings/WindowsMdmPage/WindowsMdmPage.tsx

```diff
@@ -30,15 +31,25 @@ const useSetWindowsMdm = ({
 
   const turnOnWindowsMdm = async () => {
     try {
-      const updatedConfig = await configAPI.update({
-        mdm: {
-          windows_enabled_and_configured: enable,
-        },
+      const updatedConfig = await configAPI.updateMDMConfig({
+        windows_enabled_and_configured: enable,
       });
       setConfig(updatedConfig);
       renderFlash("success", successMessage);
-    } catch {
-      renderFlash("error", errorMessage);
+    } catch (e) {
+      const axiosError = e as AxiosError;
+      switch (axiosError.response?.status) {
+        case 422:
+          renderFlash(
+            "error",
+            enable
+              ? "Couldn't turn on Windows MDM. Please configure Fleet with a certificate and key pair first."
```

Comment:
Just curious, does that mean the server's message is not used? So if the copy was to be changed, it would need to be changed in both backend and frontend?

Reply:
As currently implement, yes, although I may have misunderstood the discussion thread. Was this [message](https://github.com/fleetdm/fleet/issues/14446#issuecomment-1789669779) also been implemented on the backend?

Reply:
It is, returned as per the "standard" error message payload returned when there's a 422 response (I'm pretty sure we do extract those kinds of server-sent error messages elsewhere in the frontend?): https://github.com/fleetdm/fleet/pull/14858/files#diff-ff669b9f96ea80679f4651e9cf45ded57d5cd939d1e4e24977eb72d37d71e8bcR733

Reply:
Thanks! I'll tweak this a bit to extract the error message.